components/List/tests/snapshot/MDCSelfSizingStereoCellSnapshotTests.m

```diff
@@ -109,78 +124,111 @@ - (void)generateSnapshotWithContentSizeCategoryAndNotificationPost:
-                                                    andVerifyForView:self.cell];
+                                                    andVerifyForView:self.collectionView];
   }
 }
 
 - (void)testCellWithDynamicTypeForContentSizeCategoryExtraLargeEnabledForTitleAndDetail {
   if (@available(iOS 10.0, *)) {
     // Given
+    MDCSelfSizingStereoCell *cell = [[MDCSelfSizingStereoCell alloc] init];
     self.typographyScheme =
         [[MDCTypographyScheme alloc] initWithDefaults:MDCTypographySchemeDefaultsMaterial201902];
 
     // When
-    self.cell.titleLabel.text = @"Title";
-    self.cell.titleLabel.font = self.typographyScheme.subtitle1;
-    self.cell.detailLabel.text = @"Detail";
-    self.cell.detailLabel.font = self.typographyScheme.button;
-    self.cell.mdc_adjustsFontForContentSizeCategory = YES;
-    [self.cell setNeedsLayout];
+    cell.titleLabel.text = @"Title";
+    cell.titleLabel.font = self.typographyScheme.subtitle1;
+    cell.detailLabel.text = @"Detail";
+    cell.detailLabel.font = self.typographyScheme.button;
+    cell.mdc_adjustsFontForContentSizeCategory = YES;
+    self.arrayOfCells = @[ cell ];
 
     // Then
     [self generateSnapshotWithContentSizeCategoryAndNotificationPost:UIContentSizeCategoryExtraLarge
-                                                    andVerifyForView:self.cell];
+                                                    andVerifyForView:self.collectionView];
   }
 }
 
+- (NSInteger)numberOfSectionsInCollectionView:(UICollectionView *)collectionView {
+  return 1;
+}
+
+- (NSInteger)collectionView:(UICollectionView *)collectionView
+     numberOfItemsInSection:(NSInteger)section {
+  return self.arrayOfCells.count;
+}
+
+- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView
+                  cellForItemAtIndexPath:(NSIndexPath *)indexPath {
+  MDCSelfSizingStereoCell *dequeuedCell =
+      [collectionView dequeueReusableCellWithReuseIdentifier:kSelfSizingStereoCellIdentifier
+                                                forIndexPath:indexPath];
+  // Temporarily add dequeuedCell as a subview of collectionView
+  // to inherit traitCollection. The dequeuedCell will be removed
+  // from parent at the end of this method.
+  [self.collectionView addSubview:dequeuedCell];
+  MDCSelfSizingStereoCell *cell = self.arrayOfCells[indexPath.item];
+
+  // Change backgroundColor to distinguish cell from the collection view.
+  dequeuedCell.backgroundColor = UIColor.whiteColor;
+  dequeuedCell.titleLabel.text = cell.titleLabel.text;
+  dequeuedCell.titleLabel.font = cell.titleLabel.font;
+  dequeuedCell.detailLabel.text = cell.detailLabel.text;
+  dequeuedCell.detailLabel.font = cell.detailLabel.font;
+  dequeuedCell.leadingImageView.image = cell.leadingImageView.image;
+  dequeuedCell.trailingImageView.image = cell.trailingImageView.image;
+  dequeuedCell.mdc_adjustsFontForContentSizeCategory = cell.mdc_adjustsFontForContentSizeCategory;
+  [dequeuedCell removeFromSuperview];
```

Comment:
If MDCSelfSizingStereoCell implemented `-traitCollectionDidChange:` to call `-adjustFontsForDynamicType` similar to how `-contentSizeCategoryDidChange:` does it then we wouldn't have to do this roundabout adding and removing of subviews. I think that might be an improvement. It would result in the test code more closely resembling app code. Right now this "cellForItemAt" implementation is sort of unusual. But this is also fine.